File: DI/Select-persona-Azure/app/domain/interfaces/repositories/IRepositoryDepartamentos.ts
import { Departamento } from '../../entities/Departamento';

export interface IRepositoryDepartamentos {
  getListaDepartamentos(): Promise<Departamento[]>;
  crearDepartamento(departamentoNuevo: Departamento): Promise<number>;
  actualizarDepartamento(idDepartamento: number, departamento: Departamento): Promise<number>;
  eliminarDepartamento(idDepartamento: number): Promise<number>;
  getDepartamentoById(idDepartamento: number): Promise<Departamento | null>;
}

File: ListadoPersonasYDepartamentos/Domain/Repositories/IRepositoryDepartamentos.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Repositories
{
    public interface IRepositoryDepartamentos
    {
        Departamento[] getListaDepartamentos();

        int crearDepartamento(Departamento departamentoNuevo);

        int actualizarDepartamento(int idDepartamento, Departamento departamento);

        int eliminarDepartamento(int idDepartamento);

        Departamento getDepartamentoById(int idDepartamento);
    }
}
